server/events/vcs/gitea/client.go

```diff
@@ -421,13 +421,11 @@ func (c *GiteaClient) GetTeamNamesForUser(repo models.Repo, user models.User) ([
 // GetFileContent a repository file content from VCS (which support fetch a single file from repository)
 // The first return value indicates whether the repo contains a file or not
 // if BaseRepo had a file, its content will placed on the second return value
-func (c *GiteaClient) GetFileContent(logger logging.SimpleLogging, pull models.PullRequest, fileName string) (bool, []byte, error) {
-	logger.Debug("Getting file content for %s in Gitea pull request %d", fileName, pull.Num)
-
-	content, resp, err := c.giteaClient.GetContents(pull.BaseRepo.Owner, pull.BaseRepo.Name, pull.HeadCommit, fileName)
```

Comment:
With this change removing the `models.PullRequest` from the interface I opted to changing the head commit here to the branch name which seems to be supported by the docs.

https://pkg.go.dev/code.gitea.io/sdk/gitea#Client.GetContents

https://gitea.com/gitea/go-sdk/src/commit/8b855b4a5b33bc2050143edc4044ee330bf09b05/gitea/repo_file.go#L196

[The name of the commit/branch/tag. Default the repository’s default branch (usually master)](https://docs.gitea.com/api/1.22/#tag/repository/operation/repoGetContentsList)